packages/node-experimental/src/sdk/api.ts

```diff
@@ -0,0 +1,225 @@
+// PUBLIC APIS
+
+import { context } from '@opentelemetry/api';
+import { DEFAULT_ENVIRONMENT, closeSession, makeSession, updateSession } from '@sentry/core';
+import type {
+  Breadcrumb,
+  BreadcrumbHint,
+  CaptureContext,
+  Client,
+  Event,
+  EventHint,
+  EventProcessor,
+  Extra,
+  Extras,
+  Primitive,
+  Session,
+  Severity,
+  SeverityLevel,
+  User,
+} from '@sentry/types';
+import { GLOBAL_OBJ, consoleSandbox, dateTimestampInSeconds } from '@sentry/utils';
+import { getScopesFromContext, setScopesOnContext } from '../utils/contextData';
+
+import type { ExclusiveEventHintOrCaptureContext } from '../utils/prepareEvent';
+import { parseEventHintOrCaptureContext } from '../utils/prepareEvent';
+import type { Scope } from './scope';
+import { getClient, getCurrentScope, getGlobalScope, getIsolationScope } from './scope';
+
+export { getCurrentScope, getGlobalScope, getIsolationScope, getClient };
+export { setCurrentScope, setIsolationScope } from './scope';
+
+/**
+ * Fork a scope from the current scope, and make it the current scope in the given callback
+ */
+export function withScope<T>(callback: (scope: Scope) => T): T {
+  return context.with(context.active(), () => callback(getCurrentScope()));
+}
+
+/**
+ * Fork a scope from the current scope, and make it the current scope in the given callback.
+ * Additionally, also setup a new isolation scope for the callback.
+ */
+export function withIsolationScope<T>(callback: (scope: Scope, isolationScope: Scope) => T): T {
+  const ctx = context.active();
+  const currentScopes = getScopesFromContext(ctx);
+  const scopes = currentScopes
+    ? { ...currentScopes }
+    : {
+        scope: getCurrentScope(),
+        isolationScope: getIsolationScope(),
+      };
+
+  scopes.isolationScope = scopes.isolationScope.clone();
+
+  return context.with(setScopesOnContext(ctx, scopes), () => {
+    return callback(getCurrentScope(), getIsolationScope());
+  });
+}
```

Comment:
We decided to fork the isolation scope, because otherwise you may loose data. Think of this realistic scenario:

```js
Sentry.init();

Sentry.setTag('xx'); // <-- this actually writes on the initial isolation scope

app.get('/my-route', function() {
  // this has a forked isolation scope - but it has inherited the xx tag from outside!
  // so the tag is still present on stuff in here
}
```

Does that make sense?

For when to use it, this should mostly be used by ourselves in auto instrumentation, or if users want to do that themselves. Alternatively you can also use `setIsolationScope()`, which overwrites the isolation scope for the current execution context - whatever works better for the concrete use case.

Reply:
Ok I guess that makes sense. I was under the impression that `Sentry.setTag('xx')` would write to the global scope and that we'd only create the first isolation scope when a request hits `/my-route`. Hence my question.

Anyway, my main concern here is avoiding bleed of e.g. user data between isolation scopes which is why I had mixed feelings about forking it.